content/300-guides/300-upgrade-guides/200-upgrading-versions/050-upgrading-to-prisma-3/index.mdx

```diff
@@ -0,0 +1,78 @@
+---
+title: 'Upgrade to Prisma 3'
+metaTitle: 'Upgrade to Prisma 3'
+metaDescription: 'Guides on how to upgrade to Prisma 3'
+tocDepth: 2
+toc: true
+---
+
+<TopBlock>
+
+Prisma 3 introduces a number of **breaking changes** if you are upgrading from an earlier version (any version pre 3.x), therefore, it is important to understand how this upgrade might affect your application.
+
+Below you will find a list of the breaking changes and how to handle them.
+
+</TopBlock>
+
+## Breaking changes
+
+There are a number of breaking changes to be aware of when upgrading from version 2.x to 3.x.
+
+### [Referential actions](../../../../concepts/components/prisma-schema/relations/referential-actions)
+
+The introduction of referential actions in version 3.x removes the safety net in the Prisma Client that had previously prevented cascading deletes at runtime.
+
+As a result, depending on which workflow you are using to work on your application, you could be impacted. We advise you to check your schema and decide if you need to define referential actions explicitly.
+
+See [Referential action upgrade paths](./referential-actions) to understand how to proceed.
+
+### [$queryRaw](../../../../concepts/components/prisma-client/raw-database-access)
+
+From version 3.x onwards, the `$queryRaw` method now only supports a template literal.
+
+This means that if your application relied on `$queryRaw` calls using _strings_, those calls will **not** work anymore. We advise you to use template literals wherever possible for security reasons or resort to `$queryRawUnsafe` otherwise, after carefully escaping queries to prevent SQL injections.
+
+You can learn more about the new `$queryRaw` and `$queryRawUnsafe` methods in the [Raw database access](../../../../concepts/components/prisma-client/raw-database-access) section of the docs.
+
+### [Named constraints](../../../../concepts/components/prisma-schema/names-in-underlying-database)
+
+We changed the convention followed by Prisma to name constraints and indexes. We also introduced a `map` attribute in the PSL to allow defining these names explicitly from the Prisma schema.
```

Comment:
```suggestion
We changed the convention followed by Prisma to name constraints and indexes. We also introduced a clear distinction between the `map` attribute (database-level name) and `name` attribute (Prisma client API name) in the PSL to explicitly control how constraints are defined in the Prisma schema.
```
I realized that `map` and `name` both play a role so it should be clear even here